contrib/mesos/pkg/queue/historical.go

```diff
@@ -352,30 +351,22 @@ func (f *HistoricalFIFO) gc() {
 // Assumes that the caller has acquired the state lock.
 func (f *HistoricalFIFO) merge(id string, obj UniqueCopyable) (notifications []Entry) {
 	item, exists := f.items[id]
-	now := time.Now()
-	if !exists {
+	if !exists || item.Is(POP_EVENT|DELETE_EVENT) {
```

Comment:
I'm mostly sure that this change won't break anything since an object's UID isn't supposed to change over time, and all of our indexing is based on that UID